terraform/launchpad-token-dispenser-lambda.tf

```diff
@@ -43,4 +43,130 @@ resource "aws_ssm_parameter" "launchpad_token_dispenser_lambda_arn" {
   name  = "/service/token-dispenser/${var.prefix}"
   type  = "String"
   value = aws_lambda_function.launchpad_token_dispenser_lambda.arn
-}
\ No newline at end of file
+}
+
+resource "aws_cloudwatch_log_group" "tds_cloudtrail_log_group" {
+  name              = "/aws/cloudtrail/${var.prefix}-launchpad-token-dispenser"
+  retention_in_days = var.log_retention_days
+}
+
+resource "aws_iam_role" "tds_cloudtrail_to_cloudwatch_role" {
+  name = "${var.prefix}-cloudtrail-to-cloudwatch-role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Effect = "Allow",
+        Principal = {
+          Service = "cloudtrail.amazonaws.com"
+        },
+        Action = "sts:AssumeRole"
+      }
+    ]
+  })
+}
+
+# Inline IAM Policy for CloudTrail to write to CloudWatch Logs
+resource "aws_iam_role_policy" "tds_cloudtrail_to_cloudwatch_policy" {
+  name = "${var.prefix}-cloudtrail-to-cloudwatch-policy"
+  role = aws_iam_role.tds_cloudtrail_to_cloudwatch_role.name
+
+  policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Effect = "Allow",
+        Action = [
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ],
+        Resource = [
+          "${aws_cloudwatch_log_group.tds_cloudtrail_log_group.arn}:*"
+        ]
+      }
+    ]
+  })
+}
+
+resource "aws_cloudtrail" "launchpad_token_dispenser_trail" {
+  depends_on = [ aws_cloudwatch_log_group.tds_cloudtrail_log_group ]
+  name                          = "${var.prefix}-launchpad-token-dispenser-trail"
+  s3_bucket_name                = aws_s3_bucket.cloudtrail_bucket.id
+  include_global_service_events = true
+  is_multi_region_trail         = true
+  enable_logging                = true
+
+  # Bind cloudtrail to the CloudWatch Logs group
+  cloud_watch_logs_group_arn = "${aws_cloudwatch_log_group.tds_cloudtrail_log_group.arn}:*"
+  cloud_watch_logs_role_arn  = aws_iam_role.tds_cloudtrail_to_cloudwatch_role.arn
+
+  event_selector {
+    read_write_type           = "All"
+
+    data_resource {
+      type   = "AWS::Lambda::Function"
+      values = [aws_lambda_function.launchpad_token_dispenser_lambda.arn]
+    } 
+  }
+}
+
+
+# S3 bucket for CloudTrail logs
+resource "aws_s3_bucket" "cloudtrail_bucket" {
+  bucket = "${var.prefix}-cloudtrail-logs"
+
+  acl = "private"
```

Comment:
The `acl` arg is depreciated and private is already the default - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket#acl-1